### DISCUSSION TITLE
Accessing VERCEL_URL || NEXT_PUBLIC_VERCEL_URL in client

### DISCUSSION THREAD

**Discussion by atawfique21** [vercel/next.js#27863] [Help] [answered]
Hi there, 

I am trying to access either process.env.VERCEL_URL or process.env.NEXT_PUBLIC_VERCEL_URL inside of my client code. 

I've referred to discussions on this exact topic but it seems that Vercel has changed `Environment Variables` inside of `Project Settings` in the last few months and I cannot use the guides previous provided such as this: https://github.com/vercel/next.js/discussions/15424#discussioncomment-269353.

I have tried to do the following as shown in this discussion: https://github.com/vercel/next.js/discussions/16429#discussioncomment-797761. 

![Screen Shot 2021-08-08 at 1 03 01 PM](https://user-images.githubusercontent.com/21340196/128639681-16b20100-3a7a-4f46-a3a5-eea32e988bee.png)
And along with this, followed this guide here: https://github.com/vercel/next.js/discussions/15424#discussioncomment-41284 and created a `.env.production` file but it also returned **undefined**

Since I can no longer create environment variables that references system environment variables, how do I access them in the client code?

Thank you!

**Comment by leerob** [comment, marked as answer]
This should still work.

```js
# .env.production
NEXT_PUBLIC_VERCEL_URL=${VERCEL_URL}
```

Could you share a repro of it being `undefined`?

**Reply by atawfique21** [reply]
After doing a bunch more debugging on this, I figured out that pointing `NEXT_PUBLIC_VERCEL_URL` to `VERCEL_URL` inside of `Project Settings` is **the wrong approach.** This does not work and I'll be leaving a message in the thread so future comers do not do this. 

Since I've removed that, I've made a bit of progress by doing this
`NEXT_PUBLIC_VERCEL_URL=${VERCEL_URL}` inside of `.env.production`

~~The weird thing is, `http:REDACTED` is what is being returned by `NEXT_PUBLIC_VERCEL_URL` currently. It's missing the `//` inside of `http://`. Do you know why this is the case and how I may be able to fix this?~~

I see that Vercel_url returns without `http://` or `https://`. How can I add this to my NEXT_PUBLIC_VERCEL_URL variable? In my client code, I can do this: `https://${process.env.NEXT_PUBLIC_VERCEL_URL}/`but this is quite frustrating as my `.env.local` is `http://`, not `https://`. Thanks!

**Reply by atawfique21** [reply]
> I see that Vercel_url returns without `http://` or `https://`. How can I add this to my NEXT_PUBLIC_VERCEL_URL variable? In my client code, I can do this: `https://${process.env.NEXT_PUBLIC_VERCEL_URL}/`but this is quite frustrating as my `.env.local` is `http://`, not `https://`. Thanks!

I was able to find a solution to this by checking the `NEXT_PUBLIC_VERCEL_ENV` to be production and if so, returning `https://` on the client side code. 

For future-comers, if you want to achieve this:
```
# inside .env.production add
NEXT_PUBLIC_VERCEL_ENV=${VERCEL_ENV}
```
```
# inside .env.local add
NEXT_PUBLIC_VERCEL_ENV=development
```
```JS
// in JS,
const http = process.env.NEXT_PUBLIC_VERCEL_ENV !== 'development' ? 'https://' : 'http://'
```
